app/services/proxy_status/pulp.rb

```diff
@@ -0,0 +1,46 @@
+require 'uri'
+
+module ProxyStatus
+  class Pulp < ::ProxyStatus::Base
+    def initialize(proxy, opts = {})
+      super(proxy, opts, 'Pulp')
+    end
+
+    def storage
+      fetch_proxy_data do
+        api.pulp_storage
+      end
+    end
+
+    def status
+      body = RestClient.get(pulp_url)
+      fail _("Pulp does not appear to be running.") if body.empty?
+      json = JSON.parse(body)
+      json['errors'] = {}
+
+      if json['known_workers'].empty?
+        json['errors']['known_workers'] = _("No pulp workers running.")
+      end
+
+      if json['database_connection'] && json['database_connection']['connected'] != true
+        json['errors']['database_connection'] = _("Pulp database connection issue.")
+      end
+
+      if json['messaging_connection'] && json['messaging_connection']['connected'] != true
+        json['errors']['messaging_connection'] = _("Pulp message bus connection issue.")
+      end
+
+      json
+    end
+
+    private
+
+    def pulp_url
+       url = URI.parse(proxy.url)
+       url.port = 443
+       url.path = '/pulp/api/v2/status/'
+       url.to_s
+    end
+  end
+end
+# ::ProxyStatus.status_registry.add(ProxyStatus::Pulp)
```

Comment:
I get:
`
NoMethodError (undefined method 'status' for nil:NilClass): /home/oprazak/projects/katello/app/controllers/katello/concerns/smart_proxies_controller_extensions.rb:20:in 'pulp_status'
`
this is probably the reason. Uncommenting does not work for me, so maybe move it to Engine?